internal/topic/core.go

```diff
@@ -185,6 +186,47 @@ func (c *Core) DeleteTopic(ctx context.Context, m *Model) error {
 		}
 		return merror.New(merror.NotFound, "Topic not found")
 	}
+	// cleanup topics from broker
+	if err := c.DeleteBrokerTopic(ctx, m); err != nil {
+		return err
+	}
+	return c.repo.Delete(ctx, m)
+}
+
+// DeleteSubscriptionTopic deletes a subscription's topic and all resources associated with it
+func (c *Core) DeleteSubscriptionTopic(ctx context.Context, m *Model) error {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "TopicCore.DeleteSubscriptionTopic")
+	defer span.Finish()
+
+	topicOperationCount.WithLabelValues(env, "DeleteSubscriptionTopic").Inc()
+
+	startTime := time.Now()
+	defer func() {
+		topicOperationTimeTaken.WithLabelValues(env, "DeleteSubscriptionTopic").Observe(time.Now().Sub(startTime).Seconds())
+	}()
+
+	if ok, err := c.projectCore.ExistsWithID(ctx, m.ExtractedProjectID); !ok {
+		if err != nil {
+			return err
+		}
+		return merror.Newf(merror.NotFound, "Project not found")
+	}
+	// clean-up topic from broker
+	if err := c.DeleteBrokerTopic(ctx, m); err != nil {
+		return err
```

Comment:
if cleanup is not enabled, this will pass as deleteBrokerTopic will return nil, but data will be deleted from consul for non retry/internal topics. Is that intentional..?

Reply:
@sakshi-goyal-razorpay  yes this is intentional as config will tell if we want to delete kafka topics or not